### DISCUSSION TITLE
2.5.0-rc.1

### DISCUSSION THREAD

**Discussion by ruibaby** [halo-dev/halo#3848] [General]
## 新特性

- 提供繁体中文（TW）翻译。 by @den13501 in <https://github.com/halo-dev/halo/pull/3672>
- 对初始超级管理员用户名增加合法性校验。 by @guqing in <https://github.com/halo-dev/halo/pull/3744>
- 将文章摘要作为 Meta description 以优化文章页的 SEO。 by @guqing in <https://github.com/halo-dev/halo/pull/3745>
- 新增是否生成相对地址的配置。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/3772>
- 为访客端提供一套完整的 API。 by @guqing in <https://github.com/halo-dev/halo/pull/3787>

## 功能优化

- 允许用户修改自己的元数据信息。 by @guqing in <https://github.com/halo-dev/halo/pull/3739>
- 优化 Console 端安装和升级主题的流程，支持自动关闭安装弹框和显示反馈提示。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3777>
- 优化 Console 端激活主题的流程，支持激活后自动刷新页面。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3779>
- 优化 Console 端上传附件的弹框，支持点击完成按钮以关闭弹框。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3784>
- 优化用户账号创建流程，支持同时设置密码和角色。 by @guqing in <https://github.com/halo-dev/halo/pull/3803>
- 为 FormKit 表单中包含了 required 验证的表单元素的 label 添加指示器（*），优化可访问性。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3824>
- 优化 Console 端附件库的图片渲染性能。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3831>

## 问题修复

- 修复登录成功或失败后报错的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/3658>
- 修复 Console 端附件存储策略编辑表单无法正常渲染的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3663>
- 修复后台全局搜索中关于 Console 路由的文字没有适配 i18n 的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3665>
- 修复 Console 端 FormKit Schema 生成的表单中，Radio 输入类型值类型为 boolean 但结果为 string 的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3668>
- 修复 Console 端文章设置中的发布时间无法回显导致重新保存时丢失发布时间的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3671>
- 升级 `@halo-dev/richtext-editor` 的版本以解决编辑器部分文本翻译不完整的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3682>
- 修复保存系统设置之后导致激活主题的设置值丢失，恢复为了默认主题的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3691>
- 修复 Console 端预览主题页面的设置表单可能出现无法渲染的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3706>
- 解决反向代理后无法正确获取当前请求 URI 的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/3709>
- 修复 Console 端菜单项和菜单关联可能发现混乱的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3714>
- 修复 Console 端主题设置中，选项卡会重复显示的问题。 by @LIlGG in <https://github.com/halo-dev/halo/pull/3727>
- 修复作者页无法访问的问题。 by @guqing in <https://github.com/halo-dev/halo/pull/3736>
- 修复在 1080P 屏幕下，登录页面的样式问题。 by @hughding in <https://github.com/halo-dev/halo/pull/3756>
- 修复 Console 端仪表盘添加组件时的报错问题。 by @LIlGG in <https://github.com/halo-dev/halo/pull/3770>
- 修复文章/页面已有内容重新编辑时，无法立即显示大纲的问题。 by @LIlGG in <https://github.com/halo-dev/halo/pull/3773>
- 修复 Console 端升级插件之后没有自动刷新页面的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3781>
- 修复 Console 端插件详情页面中角色模板无法正常加载的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3794>
- 修复 Console 端的文章分类和标签选择组件在 Repeater 组件中，下拉框无法完整显示的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3819>
- 修复当插件异常时，无法正常升级插件的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/3836>
- 修复插件安装后无法启动会导致卸载后插件 JAR 文件残留的问题。 by @guqing in <https://github.com/halo-dev/halo/pull/3841>

## 依赖更新

- 升级 Spring Boot 至 3.0.6。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/3832>
- 升级 Console 端 `@tanstack/vue-query` 依赖的版本至 [v4.29.1](https://github.com/TanStack/query/releases/tag/v4.29.1) by @ruibaby in <https://github.com/halo-dev/halo/pull/3771>

## 开发者相关

- 支持在插件中定义 Reconciler。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/3789>
- 为 Console 端 FormKit 的 group 输入组件重构布局和样式。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3786>
- 为 Console 端添加 Description 组件。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3792>
- Console 端的附件选择组件支持 accepts、min、max 参数用来限定文件格式和数量。 by @ruibaby in <https://github.com/halo-dev/halo/pull/3827>

## 文档

<https://docs.halo.run>

## 资源

适用于 2.0 的插件和主题可查阅：<https://github.com/halo-sigs/awesome-halo>

**Full Changelog**: <https://github.com/halo-dev/halo/compare/v2.4.0...v2.5.0-rc.1>

<hr /><em>This discussion was created from the release <a href='https://github.com/halo-dev/halo/releases/tag/v2.5.0-rc.1'>2.5.0-rc.1</a>.</em>

**Comment by mobeicanyue** [comment]
好多更新内容啊，期待 开发者加油💪
